ReactiveCocoa/NSObject+Association.swift

```diff
@@ -12,11 +36,35 @@ extension Reactive where Base: NSObject {
 	/// - returns:
 	///   The associated value for the specified key.
 	internal func associatedValue<T>(forKey key: StaticString = #function, initial: (Base) -> T) -> T {
-		var value = objc_getAssociatedObject(base, key.utf8Start) as! T?
+		var value = base.associatedValue(forKey: key.utf8Start) as! T?
 		if value == nil {
 			value = initial(base)
-			objc_setAssociatedObject(base, key.utf8Start, value, .OBJC_ASSOCIATION_RETAIN)
+			base.setAssociatedValue(value, forKey: key.utf8Start)
 		}
 		return value!
 	}
 }
+
+extension NSObject {
+	/// Retrieve the associated value for the specified key.
+	///
+	/// - parameters:
+	///   - key: The key.
+	///
+	/// - returns:
+	///   The associated value, or `nil` if no value is associated with the key.
+	@nonobjc internal func associatedValue(forKey key: UnsafeRawPointer) -> Any? {
```

Comment:
My thought was that it would isolate the force casting to this method. The call site would still need to declare the type, but `let foo: String = bar.associatedValue(forKey: blah)` might be nicer.

I don't feel strongly about this.